nixos/modules/services/matrix/synapse.nix

```
@@ -234,17 +293,29 @@ in {
         };
 
         tls = mkOption {
-          type = types.bool;
-          default = !workerContext;
+          type = types.nullOr types.bool;
+          default = if config.path != null then
+            null
+          else
+            !workerContext;
+          defaultText = ''
+            Enabled, unless the listener is the main instance listener, or it is configured with a UNIX domain socket path.
```

```suggestion
            Enabled for the main instance listener unless it is configured with a UNIX domain socket path.
```

Worker listeners have TLS disabled by default, the main instance listener enables TLS.

---

Applied